mne/tests/test_source_tfr.py

```diff
@@ -0,0 +1,201 @@
+from copy import deepcopy
+import os.path as op
+
+import numpy as np
+from numpy.testing import (assert_array_almost_equal, assert_array_equal,
+                           assert_allclose, assert_equal)
+import pytest
+
+from mne.utils import (_TempDir, requires_h5py, run_tests_if_main)
+from mne.source_tfr import SourceTFR
+
+
+def _fake_stfr():
+    """Create a fake SourceTFR."""
+    verts = [np.arange(10), np.arange(90)]
+    return SourceTFR(np.random.rand(100, 20, 10), verts, 0, 1e-1, 'foo')
+
+
+def _fake_kernel_stfr():
+    """Create a fake kernelized SourceTFR."""
+    kernel = np.random.rand(100, 40)
+    sens_data = np.random.rand(40, 20, 10)
+    verts = [np.arange(10), np.arange(90)]
+    return SourceTFR((kernel, sens_data), verts, 0, 1e-1, 'foo')
+
+
+def test_stfr_kernel_equality():
+    """Test kernelized and normal SourceTFR equality."""
+    kernel = np.random.rand(100, 40)
+    sens_data = np.random.rand(40, 10, 30)
+    verts = [np.arange(10), np.arange(90)]
+    data = np.tensordot(kernel, sens_data, axes=([-1], [0]))
+    tmin = 0
+    tstep = 1e-3
+
+    kernel_stfr = SourceTFR((kernel, sens_data), verts, tmin, tstep)
+    full_stfr = SourceTFR(data, verts, tmin, tstep)
+
+    # check if data is in correct shape
+    expected = [100, 10, 30]
+    assert_allclose([kernel_stfr.shape, full_stfr.shape,
```

Comment:
why are you testing it like this? by creating a vector that will have no name and when we read the error we will know nothing about it? why allclose? is it possible to have an array with shape 100+epsilon?


```py
assert kernel_stfr.shape == (100, 10, 30)
assert full_stfr.shape == (100, 10, 30)
assert bla.shape == (100, 10, 30)
assert foo.shape == (watever)
``` 
will return a meaningful error that I will read and I would understand what is my next step. otherwise, I'll have to put a breakpoint stop the execution, understand the array you have created think about all those things and make a guess of what could be wrong. I'm not smart enough. I would rather prefer pytest telling me where should I look.